### DISCUSSION TITLE
Heat flux monitoring and maxDi number

### DISCUSSION THREAD

**Discussion by torodecamargue** [nextfoam/baram#113] [Ideas]
Hello,

When using CHT cases with Baram, I think that there is 2 usefull tools missing.

Monitoring the power should be very good by doing integral of the heat flux. Do you think you could add this tools?

One last thing is that BaramFlow uses by default a maxDi number of 10. For a frozen flow calculation it is better to use higher values in order to increase the timestep value of the calculation. 
Couyld you add the possibility to change the maxDi number like it is already done for the maxCo number in the "Run Conditions" window? 

One last comment, when I do the flow calculation for the initialization of the froezn flow, I don't calculate the energy equation to save time, but the residuals view during calculation of the flow is not working.

![image](https://github.com/user-attachments/assets/1b23f13c-0c53-4338-adf8-749cc068283c)

Thank you 

Best regards

Julien

**Comment by kdg5424** [comment]
Hello Julien. This is Donggyu from NEXTFOAM.

1. Well, I should discuss with @jiban. And could you let us know why do you need it?
2. Could you tell me what is MaxDi number? Do you mean like Nusselt, Prandtl number?

In my PC, the residuals show properly for the Energy off case. So could you send your case?
Never mind the quality of the residuals. It's just for showing.
![image](https://github.com/user-attachments/assets/4b130aba-083a-45c7-88ac-42ea857e4f37)

Thanks for in advance.

Best regards
Donggyu kim

**Comment by torodecamargue** [comment]
Hello Donggyu,

1. I think it is very important to have a monitoring of the heat fluxes. You can know how much power is absorbed by solid surfaces and optimize the design of the hot air convection duct. Maybe it is possible to use wallHeatFlux function, I don't know if it is compatible with BaramFlow? Something to add in the controlDict file, I can try.

wallHeatFlux1
{
    type        wallHeatFlux;
    libs        ("libfieldFunctionObjects.so");
    patches     ("solid_to_fluid");
    region      solid;
    executeControl  writeTime;
    writeControl    writeTime;
}



2. The parameter Di is the diffusion number in solid. It is comparable to the courant number in fluid. This number acts on the timestep value in transient calculations and by default max Di = 10. This parameter is declared in the controlDict file.


In the case below, when your start the calculation the monitoring is not displayed
[calcul_CHT_cyl_steel - v4ms_steady-kwSST_y10 - forum.zip](https://github.com/user-attachments/files/16657611/calcul_CHT_cyl_steel.-.v4ms_steady-kwSST_y10.-.forum.zip)

Best regards

**Comment by kdg5424** [comment]
Hello Julien. This is Donggyu.

1. I understand what you mean. I'll discuss with our developers the level of implementation. And we'll decide to add it after discussion. Thank you for your suggestion.

2. If it's not difficult to implement and if it's really necessary, let us consider to add it.

3. I think the problem is from **CHT**. As you know, almost (99% over) simulation with solid solves heat transfer.
So I think if the users solve without energy equation, the code casues bug. We'll fix it as soon as possible.

Thanks

Regards

Donggyu kim

**Comment by kdg5424** [comment]
And one more thing, I recommend you not to make blank in the project name.

**Comment by torodecamargue** [comment]
Hello Donggyu,

In adding the function wallHeatFlux on the solid surface in the control dict file it works fine. I am able to postprocess the heat flux in [W], I think it could be quite simple for your team to add this in the monitoring choices. I hope you could do it in the next release.

I add this text in the controlDict file:

 wallHeatFlux1
{
    type        wallHeatFlux;
    libs        ("libfieldFunctionObjects.so");
    patches     ("wall_solid");
    region      solid;
    writeFields false;
    writeControl timeStep;
    writeInterval 1;


Than you very much for your work

Julien

**Comment by kdg5424** [comment]
Hello Julien. Thank you for your suggestion.
We'll consider it.

Donggyu kim

**Comment by jiban** [comment]
Hi Julien,
BARAM v24.4.0 now has an input for `maxDi`(*Max Diffusion Number*) when a multi-regions case is configured for transient calculation.

![image](https://github.com/user-attachments/assets/134b47e4-6e6e-48d8-bef6-7a459c0a97bc)

Please try this.
Thank you.

**Comment by torodecamargue** [comment]
Hello jiban,

Thank you I will try this soon!

Julien

**Comment by torodecamargue** [comment]
Hello,

Do you think that there is a chance to have the Heat flux monitoring or in reports section ("reports" is really helpfull feature! ) in the next release of BaramFlow?

Best regards

Julien

**Comment by jiban** [comment]
Hi Julien,
Wall Heat Flux report will be included in the next release, which will be available in around a month.
Thank you for your patience in the long waiting.
Happy new year!

-- Jake

**Comment by jiban** [comment]
Hi Julien,
I'm happy to let you know that reporting for *Wall Heat Flux* and *Heat Transfer Coefficient* is available in v25.1.0.


![image](https://github.com/user-attachments/assets/c136d658-f38c-4b0b-b9e2-862e00448f38)


Yet it only calculates and writes the fields on the disk.
You can see the fields only with paraview or other 3rd party viewer since BARAM does not provide post-processing feature for now.

Thank you.

-- Jake
